app/services/optout/OptOutService.scala

```diff
@@ -150,7 +149,7 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
     repository.set(data)
   }
 
-  def makeOptOutUpdateRequest(optOutProposition: OptOutProposition)
+  def makeOptOutUpdateRequest(optOutProposition: OptOutProposition, intentFuture: Future[Option[TaxYear]])
```

Comment:
I understand passing Future as method argument isn't within our coding standards, is that correct @avelx

Reply:
just thinking, how about passing 'intentCallback: () => Future[Option[TaxYear]]' instead?

Reply:
I mentioned in the PR description that I'm not a big fan of this approach and would like to get to a cleaner implementation, but have been hampered by the complexity of the for comprehension. I'd rather do this by (flat)mapping the result of getting the customer intent into a function that would then make the ITSA status upsert requests in that context, but we're short on time to fix the E2E testing issue so I'm proposing this as a pragmatic solution for now.

Reply:
approved